src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -537,15 +550,16 @@ private void UpdateCacheIfNeeded(long position, INamedOnnxValueGetter[] srcNamed
                         outputCache.Outputs[outputNameOnnxValue.Name] = outputNameOnnxValue;
```

Comment:
In this loop we also need to dispose of the old value before we just overwrite it. I changed mine to add this in the loop before the reasignment.
```
if(outputCache.Outputs.TryGetValue(outputNameOnnxValue.Name, out DisposableNamedOnnxValue value))
{
    value.Dispose();
}
```
I was hoping to avoid the extra check, but the very first time we are here there are no values in the dictionary so it fails. If there was a way we could add dummy values when we initialize the dictionary we can avoid it. This drops the leaked memory by a lot.